src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -366,19 +348,29 @@ public EvaluatedDlsFlsConfig getDlsFls(User user, IndexNameExpressionResolver re
             Map<String, Set<String>> dlsQueriesByIndex = new HashMap<String, Set<String>>();            
             Map<String, Set<String>> flsFields = new HashMap<String, Set<String>>();
             Map<String, Set<String>> maskedFieldsMap = new HashMap<String, Set<String>>();
-                        
+
+            // we capture all concrete indices that do not have any
+            // DLS/FLS/Masked Fields restrictions. If the dmf_empty_overwrites_all
+            // switch is enabled, this trumps any restrictions on those indices
+            // that may be imposed by other roles.
+            Set<String> noDlsConcreteIndices = new HashSet<>();
+            Set<String> noFlsConcreteIndices = new HashSet<>();
+            Set<String> noMaskedFieldConcreteIndices = new HashSet<>();
+
             for (SecurityRole role : roles) {
                 for (IndexPattern ip : role.getIpatterns()) {
 					Set<String> concreteIndices;
-					concreteIndices = ip.getResolvedIndexPattern(user, resolver, cs);
+					concreteIndices = ip.getResolvedIndexPattern(user, resolver, cs, false);
 					String dls = ip.getDlsQuery(user);
 
 					if (dls != null && dls.length() > 0) {
 
 						for (String concreteIndex : concreteIndices) {
 							dlsQueriesByIndex.computeIfAbsent(concreteIndex, (key) -> new HashSet<String>()).add(dls);
 						}
-					}
+					} else if (dfmEmptyOverwritesAll) {
+                        noDlsConcreteIndices.addAll(concreteIndices);
+                    }
```

Comment:
indentation seems off here

Reply:
Nit; we should add formatting linter that can auto-fix this

Reply:
I've created [FEATURE] Add java checkstyle / formatter https://github.com/opensearch-project/security/issues/1752 to track this.